sky/cli.py

```diff
@@ -3700,6 +3700,61 @@ def spot_logs(name: Optional[str], job_id: Optional[int], follow: bool,
         sys.exit(1)
 
 
+@spot.command('dashboard', cls=_DocumentedCodeCommand)
+@click.option(
+    '--port',
+    '-p',
+    default=None,
+    type=int,
+    required=False,
+    help=('Local port to use for the dashboard. If None, a free port is '
+          'automatically chosen.'))
+@usage_lib.entrypoint
+def spot_dashboard(port: Optional[int]):
+    """Opens a dashboard for spot jobs (needs controller to be UP)."""
+    # TODO(zongheng): ideally, the controller/dashboard server should expose the
+    # API perhaps via REST. Then here we would (1) not have to use SSH to try to
+    # see if the controller is UP first, which is slow; (2) not have to run SSH
+    # port forwarding first (we'd just launch a local dashboard which would make
+    # REST API calls to the controller dashboard server).
+    click.secho('Checking if spot controller is up...', fg='yellow')
+    hint = (
+        'Dashboard is not available if spot controller is not up. Run a spot '
+        'job first, or use `sky start` to bring up an existing controller.')
+    _, handle = spot_lib.is_spot_controller_up(stopped_message=hint,
+                                               non_existent_message=hint)
+    if handle is None:
+        sys.exit(1)
+    # SSH forward a free local port to remote's dashboard port.
+    remote_port = constants.SPOT_DASHBOARD_REMOTE_PORT
+    if port is None:
+        free_port = common_utils.find_free_port(remote_port)
+    else:
+        free_port = port
+    ssh_command = (f'ssh -qNL {free_port}:localhost:{remote_port} '
+                   f'{spot_lib.SPOT_CONTROLLER_NAME}')
+    click.echo('Forwarding port: ', nl=False)
+    click.secho(f'{ssh_command}', dim=True)
+    ssh_process = subprocess.Popen(ssh_command,
+                                   shell=True,
+                                   preexec_fn=os.setsid)
+
+    webbrowser.open(f'http://localhost:{free_port}')
+    click.secho(f'Dashboard is now available at: http://127.0.0.1:{free_port}',
+                fg='green')
+    try:
+        while True:
+            time.sleep(1)
```

Comment:
nit: How about actively exit the function if the ssh port forwarding finishes (like the spot controller being autostopped)
```suggestion
        while ssh_process.poll() is not None:
            time.sleep(1)
```

Reply:
This seems to immediately go into `finally` and print `Exiting.`

Reply:
Oh, how about replacing the whole while loop with `ssh_process.wait()`?

Reply:
Thanks! Works nicely with autostopped. Prints `Exiting` and verified that the port gets cleaned up.